pkg/controller/controller_utils.go

```diff
@@ -542,6 +544,39 @@ func GetPodFromTemplate(template *v1.PodTemplateSpec, parentObject runtime.Objec
 	return pod, nil
 }
 
+func (r RealPodControl) validateNodeConflict(pod *v1.Pod) error {
+	if len(pod.Spec.NodeName) == 0 {
+		return nil
+	}
+
+	node, err := r.KubeClient.Core().Nodes().Get(pod.Spec.NodeName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	nodeInfo := schedulercache.NewNodeInfo()
+	err = nodeInfo.SetNode(node)
+	if err != nil {
+		return err
+	}
+
+	fit, reasons, err := predicates.PodSelectorMatches(pod, nil, nodeInfo)
```

Comment:
Thank you folks for suggestions!
As is pointed out by Klaus, controller may not have enough info of the rejected pod. After kubelet rejects a pod, the pod will then be set in `Failed` phase and be killed (by kubelet?). Do controller know this flow? Will dig into this.